cardano-cli/src/Cardano/CLI/Byron/Parsers.hs

```diff
@@ -721,3 +719,47 @@ cliParseTxId :: String -> TxId
 cliParseTxId =
   either (panic . ("Bad Lovelace value: " <>) . show) identity
   . decodeHash . Text.pack
+
+parseFraction :: String -> String -> Parser Rational
+parseFraction optname desc =
+  option (toRational <$> readDouble) $
+      long optname
+   <> metavar "DOUBLE"
+   <> help desc
+
+parseIntegral :: Integral a => String -> String -> Parser a
+parseIntegral optname desc = option (fromInteger <$> auto)
+  $ long optname <> metavar "INT" <> help desc
+
+parseLovelace :: String -> String -> Parser Lovelace
+parseLovelace optname desc =
+  either (panic . show) identity . mkLovelace
+    <$> parseIntegral optname desc
+
+readDouble :: ReadM Double
+readDouble = do
+  f <- auto
+  when (f < 0) $ readerError "fraction must be >= 0"
+  when (f > 1) $ readerError "fraction must be <= 1"
+  return f
+
+parseFilePath :: String -> String -> Parser FilePath
+parseFilePath optname desc =
+  strOption
+    ( long optname
+    <> metavar "FILEPATH"
+    <> help desc
+    <> completer (bashCompleter "file")
+    )
+
+parseSigningKeyFile :: String -> String -> Parser SigningKeyFile
+parseSigningKeyFile opt desc = SigningKeyFile <$> parseFilePath opt desc
+
+
+parseGenesisFile :: String -> Parser GenesisFile
+parseGenesisFile opt =
+  GenesisFile <$> parseFilePath opt "Genesis JSON file."
+
+parseFlag' :: a -> a -> String -> String -> Parser a
+parseFlag' def active optname desc =
+  flag def active $ long optname <> help desc
```

Comment:
What's the significance of the prime suffix?

Reply:
:shrug:....I didn't implement this function, but I'm removing it because its only used in one place.